### PR TITLE
initial commit backend-api-routes_products

### PR DESCRIPTION
**Pull request by maria-tsek**
--Create an api route which returns all products.
--Create an api route which returns a single product based on the id passed as query parameter.
--creating an listing page with all the products. 

--Also create the detail page

### CODE REVIEW

**Review by reverbePrintemps — CHANGES_REQUESTED** [review]
Left a few comments. Otherwise looking ok 👍

File: backend/backend-api-routes/backend-api-routes_products/pages/products/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <div>
+      <h1>Products</h1>
+      <ul>
+        {data.map((product) => (
+          <li key={product.id}>
+            <h2>{product.name}</h2>
+            <p>{product.description}</p>
+            <p>Price: ${product.price}</p>
+            <p>{product.currency}</p>
```
[the remark this answers is not in the document]
Comment (maria-tsek) [reply backend/backend-api-routes/backend-api-routes_products/pages/products/index.js:20]: hahahaha , i didn't notice it 😱

File: backend/backend-api-routes/backend-api-routes_products/pages/products/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import useSWR from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
+
+function ProductsPage() {
+  const { data, error } = useSWR("/api/products", fetcher);
+
+  if (error) return <div>Error loading data</div>;
+  if (!data) return <div>Loading...</div>;
```
[the remark this answers is not in the document]
Comment (maria-tsek) [reply backend/backend-api-routes/backend-api-routes_products/pages/products/index.js:9]: you mean to use try...catch ? or something else ?

[the remark this answers is not in the document]
Comment (maria-tsek) [reply backend/backend-api-routes/backend-api-routes_products/pages/products/index.js:9]: and also this : if (!data) return <div>Loading...</div>; does not work like that ?

**Pull request merged**